itl/command-plugins.conf

```diff
@@ -1768,6 +1768,10 @@ object CheckCommand "snmp" {
 			value = "$snmp_miblist$"
 			description = "List of MIBS to be loaded (default = none if using numeric OIDs or 'ALL' for symbolic OIDs.)"
 		}
+		"-M" = {
+			value = "$snmp_multiplier$"
+			description = "Multiplies current value, 0 < n < 1 works as divider, defaults to 1"
```

Comment:
Please either use `set_if` instead of `value` here or set the `vars.snmp_multiplier` custom variable at the end of the command definition to its default value (`1`), otherwise Icinga 2 may just pass it with an empty string (`-M ''`).

Reply:
I do not think this is how it works, empty variables don't get passed and neither does the key preceeding them.

Reply:
You are right! I just had such a command in my mind `/etc/icinga2/scripts/mail-service-notification.sh' '-4' '127.0.0.1' '-6' '::1' '-X' '' '-b' '' '-c' '' '-e' 'hallo' '-l' 'master1' '-n' 'master1' '-o'`, but that doesn't seem to be the case with check commands..

Reply:
It is actually the same with check commands as well if you explicitly set them to an empty string:
```bash
...
vars.snmp_multiplier = ""
```

results to this: `/icinga2/prefix/etc/icinga2/check_snmp' '-M' ''` where `set_if` would've omitted it.

Reply:
If `snmp_multiplier` is not set, the attribute won't be passed.

This is the same behaviour as with e.g.  snmp_offset or snmp_retries